incubator/kafka/values.yaml

```diff
@@ -143,52 +143,86 @@ persistence:
-      lowercaseOutputName: true
-      jmxUrl: service:jmx:rmi:///jndi/rmi://127.0.0.1:5555/jmxrmi
-      ssl: false
-      whitelistObjectNames: ["kafka.server:*", "kafka.controller:*", "java.lang:*"]
-      # rules:
-      #   - pattern : kafka.server<type=ReplicaFetcherManager, name=MaxLag, clientId=(.+)><>Value
-      #   - pattern : kafka.server<type=BrokerTopicMetrics, name=(BytesInPerSec|BytesOutPerSec|MessagesInPerSec), topic=(.+)><>OneMinuteRate
-      #   - pattern : kafka.server<type=KafkaRequestHandlerPool, name=RequestHandlerAvgIdlePercent><>OneMinuteRate
-      #   - pattern : kafka.server<type=Produce><>queue-size
-      #   - pattern : kafka.server<type=ReplicaManager, name=(PartitionCount|UnderReplicatedPartitions)><>(Value|OneMinuteRate)
-      #   - pattern : kafka.server<type=controller-channel-metrics, broker-id=(.+)><>(.*)
-      #   - pattern : kafka.server<type=socket-server-metrics, networkProcessor=(.+)><>(.*)
-      #   - pattern : kafka.server<type=Fetch><>queue-size
-      #   - pattern : kafka.server<type=SessionExpireListener, name=(.+)><>OneMinuteRate
-      #   - pattern : kafka.controller<type=KafkaController, name=(.+)><>Value
-      #   - pattern : java.lang<type=OperatingSystem><>SystemCpuLoad
-      #   - pattern : java.lang<type=Memory><HeapMemoryUsage>used
-      #   - pattern : java.lang<type=OperatingSystem><>FreePhysicalMemorySize
+    ## Interval at which Prometheus scrapes metrics
+    interval: 10s
+
+    ## Port jmx-exporter exposes Prometheus format metrics to scrape
+    port: 5556
+
+    ## Rules to apply to the Kafka JMX Exporter
+    configMap:
+      ## Allows disabling the default configmap, note a configMap is needed
+      enabled: true
+      ## Allows setting values to generate confimap
+      overrideConfig: {}
+        # jmxUrl: service:jmx:rmi:///jndi/rmi://127.0.0.1:5555/jmxrmi
+        # lowercaseOutputName: true
+        # lowercaseOutputLabelNames: true
+        # ssl: false
+      ## Allows setting the name of the ConfigMap to use, note a configMap
+      ## with said name needs to exists for jmx stats to work
+      overrideName: ""
+    resources: {}
+      # limits:
+      #   cpu: 200m
+      #   memory: 1Gi
+      # requests:
+      #   cpu: 100m
+      #   memory: 100Mi
 
   kafka:
     enabled: false
 
-    # The image to use for the metrics collector
+    ## The image to use for the metrics collector
     image: danielqsj/kafka-exporter
 
-    # The image tag to use for the metrics collector
+    ## The image tag to use for the metrics collector
     imageTag: v1.0.1
 
+    ## Interval at which Prometheus scrapes metrics
+
+    interval: 10s
+
+    ## Port kafka-exporter exposes for Prometheus to scrape metrics
+    port: 9308
+
+    ## Resource limits
+    resources: {}
+#      limits:
+#        cpu: 200m
+#        memory: 1Gi
+#      requests:
+#        cpu: 100m
+#        memory: 100Mi
+
+  ## Are you using Prometheus Operator?
+  prometheusOperator: true
```

Comment:
Default to false

Reply:
Prometheus Operator is the standard that's recommended by CoreOS, I feel it should be defaulted to yes IMHO

Reply:
I'm also not a fan of this defaulting as `true`; even though it's recommended by CoreOS, IIRC CoreOS isn't the owner/maintainer of any of the components of this (Kubernetes, Prometheus, or Helm)

It's also not the default way that many Helm users would get prometheus (the default `prometheus` chart doesn't use the operator), so does not seem like it should be the default in this sense.

Reply:
IMHO, we can leave it true. I and many others I know of, by default use operator not the default prometheus chart. My 2 cents.

Reply:
Doesn't feel right to me to have a chart rely on a third party product and Custom Resources that don't come standard with Kubernetes.

I'm interested in seeing if e2e tests will handle it though.

Reply:
@viglesiasce I'm curious if you have a perspective on using Prometheus Operator as a default pattern in one of our charts? I am personally a fan of the operator pattern, but I wonder if perhaps its a little inconsistent to allow this as a default given that we don't offer a chart for Prometheus Operator in this repository?

I'm leaning towards defaulting this to false personally.